### PR TITLE
WIP network test using real routing

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @fraser999

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/vault.rs [hunk 1]
```diff
@@ -486,6 +486,48 @@ pub type ResponseNotifier =
...
+    #[test]
+    fn network_test() {
+        let run_vault = |mut vault: Vault| {
+            let _ = ::std::thread::spawn(move || {
+                vault.do_run();
+            });
+        };
+        for i in 0..4 {
+            println!("starting node {:?}", i);
+            let _ = run_vault(Vault::new());
```
Comment (dirvine) [inline src/vault.rs]: I think here you want to spawn these on a thread

File: src/vault.rs [hunk 2]
```diff
@@ -486,6 +486,48 @@ pub type ResponseNotifier =
...
+            println!("client routing received an event");
+            match event {
+                // Event::Request{ request, our_authority, from_authority, response_token } =>
+                //     println!("client received a request"),
+                // Event::Response{ response, our_authority, from_authority } =>
+                //     println!("client received a Response"),
+                Event::Refresh(_type_tag, _group_name, _accounts) =>
+                    println!("client received a refresh"),
+                Event::Churn(_close_group) => println!("client received a churn"),
+                Event::Connected => {
```
Comment (benjaminbollen) [inline src/vault.rs]: a client will never connect;

**Comment by benjaminbollen** [comment]
can we first just test vault::new() ?  You should see connected and churn events on a pure vault network

**Pull request merged**